ts/components/conversation/message/message-content/quote/Quote.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { MouseEvent, useState } from 'react';
+
+import * as MIME from '../../../../../types/MIME';
+
+import { QuoteAuthor } from './QuoteAuthor';
+import { QuoteText } from './QuoteText';
+import { QuoteIconContainer } from './QuoteIconContainer';
+import styled from 'styled-components';
+import { isEmpty } from 'lodash';
+
+const StyledQuoteContainer = styled.div`
+  min-width: 300px; // if the quoted content is small it doesn't look very good so we set a minimum
+  padding-right: var(--margins-xs);
+`;
+
+const StyledQuote = styled.div<{
+  hasAttachment: boolean;
+  isIncoming: boolean;
+  onClick: ((e: MouseEvent<HTMLDivElement>) => void) | undefined;
+}>`
+  position: relative;
+
+  display: flex;
+  flex-direction: row;
+  align-items: stretch;
+  margin: ${props => (props.hasAttachment ? 'var(--margins-md)' : 'var(--margins-xs)')} 0;
+  ${props => !props.hasAttachment && 'border-left: 4px solid;'}
+  border-color: ${props =>
+    props.isIncoming
+      ? 'var(--message-bubbles-received-text-color)'
+      : 'var(--message-bubbles-sent-text-color)'};
+  cursor: ${props => (props.onClick ? 'pointer' : 'auto')};
+`;
+
+const StyledQuoteTextContent = styled.div`
+  flex-grow: 1;
+  padding-inline-start: 10px;
+  padding-inline-end: 10px;
+  max-width: 100%;
+
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+`;
+
+export type QuoteProps = {
+  attachment?: QuotedAttachmentType;
+  author: string;
+  isFromMe: boolean;
+  isIncoming: boolean;
+  referencedMessageNotFound: boolean;
+  text: string | null;
+
+  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
+};
+
+export interface Attachment {
```

Comment:
we must have a bunch of already defined Attachment type. Maybe give this one a more explicit name as it is related to the Quote mostly?

Reply:
Good point I renamed it to QuotedAttachmentThumbnailType